src/interpreter/contract.rs

```diff
@@ -23,15 +23,23 @@ where
     pub(crate) fn contract_balance(&mut self, ra: RegisterId, b: Word, c: Word) -> Result<(), RuntimeError> {
         Self::is_register_writable(ra)?;
 
-        if b > VM_MAX_RAM - AssetId::LEN as Word || c > VM_MAX_RAM - ContractId::LEN as Word {
+        let bx = b
+            .checked_add(AssetId::LEN as Word)
+            .ok_or_else(|| Bug::new(BugId::ID014, BugVariant::CheckedRegisterOverflow))?;
```

Comment:
This one is not a bug but a malformed input. The bugs are reserved for things who are already checked or guaranteed by the specs to not happen. In this case, the user can just send a bad `b`

Reply:
this has been modified